src/utils.cpp

```diff
@@ -0,0 +1,38 @@
+#include "utils.h"
+#include <arpa/inet.h>
+#include <ifaddrs.h>
+#include <net/if.h>
+#include <netinet/in.h>
+
+namespace zeek {
+std::vector<std::string> getHostIPAddrs() {
+  std::vector<std::string> ip_addrs;
+
+  struct ifaddrs *ifAddrStruct = NULL;
+  struct ifaddrs *ifa = NULL;
+  void *tmpAddrPtr = NULL;
+
+  getifaddrs(&ifAddrStruct);
+
+  for (ifa = ifAddrStruct; ifa != NULL; ifa = ifa->ifa_next) {
+    if (!ifa->ifa_addr || (ifa->ifa_flags & IFF_LOOPBACK)) {
+      continue;
+    }
+
+    if (ifa->ifa_addr->sa_family == AF_INET) {
+      tmpAddrPtr = &((struct sockaddr_in *)ifa->ifa_addr)->sin_addr;
+      char addressBuffer[INET_ADDRSTRLEN];
+      inet_ntop(AF_INET, tmpAddrPtr, addressBuffer, INET_ADDRSTRLEN);
+      ip_addrs.push_back(addressBuffer);
+    } else if (ifa->ifa_addr->sa_family == AF_INET6) {
+      tmpAddrPtr = &((struct sockaddr_in6 *)ifa->ifa_addr)->sin6_addr;
+      char addressBuffer[INET6_ADDRSTRLEN];
+      inet_ntop(AF_INET6, tmpAddrPtr, addressBuffer, INET6_ADDRSTRLEN);
+      ip_addrs.push_back(addressBuffer);
+    }
+  }
+  if (ifAddrStruct != NULL)
+    freeifaddrs(ifAddrStruct);
+  return ip_addrs;
+}
```

Comment:
Let's use C++ if possible (`nullptr` instead of `NULL`; explicit casts; variables declared near first use).

We also need to handle errors from `getifaddrs` in which case we would probably log an error (left a `FIXME` for you) and return early; `freeifaddrs` would then be called unconditionally.

```suggestion
#include "utils.h"

#include <arpa/inet.h>
#include <ifaddrs.h>
#include <net/if.h>
#include <netinet/in.h>

namespace zeek {
std::vector<std::string> getHostIPAddrs() {
  std::vector<std::string> ip_addrs;

  ifaddrs *ifaddrs = nullptr;
  if (auto err = getifaddrs(&ifaddrs)) {
    // FIXME(Wajihulhassan): log error from `strerror(errno)`.
    return {};
  }

  for (auto *ifa = ifaddrs; ifa; ifa = ifa->ifa_next) {
    if (!ifa->ifa_addr || (ifa->ifa_flags & IFF_LOOPBACK)) {
      continue;
    }

    if (ifa->ifa_addr->sa_family == AF_INET) {
      auto addr =
          reinterpret_cast<struct sockaddr_in *>(ifa->ifa_addr)->sin_addr;
      char addressBuffer[INET_ADDRSTRLEN];
      inet_ntop(AF_INET, &addr, addressBuffer, INET_ADDRSTRLEN);
      ip_addrs.push_back(addressBuffer);
    } else if (ifa->ifa_addr->sa_family == AF_INET6) {
      auto addr =
          reinterpret_cast<struct sockaddr_in6 *>(ifa->ifa_addr)->sin6_addr;
      char addressBuffer[INET6_ADDRSTRLEN];
      inet_ntop(AF_INET6, &addr, addressBuffer, INET6_ADDRSTRLEN);
      ip_addrs.push_back(addressBuffer);
    }
  }

  freeifaddrs(ifaddrs);

  return ip_addrs;
}
} // namespace zeek
```